nomos-services/cryptarchia-consensus/src/leadership.rs

```diff
@@ -30,17 +30,26 @@ pub struct LeaderConfig {
 
 impl Leader {
     pub fn new(
-        genesis: HeaderId,
-        LeaderConfig { notes, nf_sk }: LeaderConfig,
+        header_id: HeaderId,
+        header_notes: Vec<NoteWitness>,
+        nf_sk: NullifierSecret,
         config: Config,
     ) -> Self {
         Leader {
-            notes: HashMap::from([(genesis, notes)]),
+            notes: HashMap::from([(header_id, header_notes)]),
             nf_sk,
             config,
         }
     }
 
+    pub fn from_genesis(
+        genesis: HeaderId,
+        LeaderConfig { notes, nf_sk }: LeaderConfig,
+        config: Config,
+    ) -> Self {
+        Leader::new(genesis, notes, nf_sk, config)
```

Comment:
My pedantic clippy suggests me this as a better alternative
```suggestion
        Self::new(genesis, notes, nf_sk, config)
```

Reply:
That slipped by, thanks.